### ISSUE TITLE
No Topics in Rosbridge with docker

### ISSUE THREAD

**Issue by frapit** [code-iai/ROSIntegration#166]
There are some issues having the same effect but I couldn't make any of the suggested solutions work.

I have unreal engine 4.27 running in the officially provided docker container. It is started with `--network=host` so it is in the same network. 

On my host system I have ubuntu 20.04 with ros noetic and started the rosbridge (tried both the downgraded version and the current version) and it successfully detects a connected client:
```
[INFO] [1651249747.115300]: [Client 0] connected. 1 client total.
```

Yet. no topic (published or subscribed) from ue4 ever shows up. I extended the example to publish a message every 2 seconds.

I also setup a container with ros melodic as many reported success using it. I have the feeling that the advertising of a topic never reaches the rosbridge. 

Any ideas? Is anyone running a docker based setup? Could the ue4 version be too recent?

EDIT
My publisher code:
```
void Aros_state_publisher::BeginPlay()
{

    Super::BeginPlay();
    ros_topic = NewObject<UTopic>(UTopic::StaticClass());

    UROSIntegrationGameInstance *rosinst = Cast<UROSIntegrationGameInstance>(GetGameInstance());
    ros_topic->Init(rosinst->ROSIntegrationCore, TEXT("/example_topic"), TEXT("std_msgs/String"));

    ros_topic->Advertise();
    UE_LOG(LogROS, Warning, TEXT("Message publisher initialized"));

    GetWorldTimerManager().SetTimer(timer_handle, this, &Aros_state_publisher::publish_ros, 1.0f,
                                    true, 2.0f);
}

void Aros_state_publisher::publish_ros()
{
    TSharedPtr<ROSMessages::std_msgs::String> StringMessage(
      new ROSMessages::std_msgs::String("Publishing a string"));
    ros_topic->Publish(StringMessage);

    UE_LOG(LogROS, Warning, TEXT("Published Message"));
}

```
and in the header:
```
  public:
    // Called every frame
    virtual void Tick(float DeltaTime) override;

    void publish_ros();
    FTimerHandle timer_handle;

    UPROPERTY()
    UTopic *ros_topic;

```

**Comment by tsender** [comment]
The problem is not with docker. It's a rosbridge connection problem, likely with ROS Noetic (as others have had similar problems).

Can you please specify the tags/branches of rosbridge you have tried? Looking through the rosbridge repo, I would think the `ros1` branch should work fine (there were several fixes made on that branch to avoid the problems you are experiencing).

**Comment by frapit** [comment]
Thank you a lot @tsender, the `ros1` branch does fix the issue!  Previously I checked the `0.11.10` tag as suggested in some other issues and I used the ros-kinetic / ros-melodic / ros-noetic and ros-foxy docker container to test the respective apt version of rosbridge_server. They all did not work.
